grid2op/Agent/alertAgent.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2019-2020, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+import numpy as np
+from grid2op.Agent.recoPowerlineAgent import RecoPowerlineAgent
+import pandas as pd
+
+#try:
+#    from lightsim2grid import LightSimBackend
+#
+#    bkclass = LightSimBackend
+#    # raise ImportError()
+#except ImportError as excq_:
+#    from grid2op.Backend import PandaPowerBackend
+
+
+
+class AlertAgent(RecoPowerlineAgent):
+    """
+    This is a :class:`AlertAgent` example, which will attempt to reconnect powerlines and send alerts on the worst possible attacks: for each disconnected powerline
+    that can be reconnected, it will simulate the effect of reconnecting it. And reconnect the one that lead to the
+    highest simulated reward. It will also simulate the effect of having a line disconnection on attackable lines and raise alerts for the worst ones
+
+    """
+
+    def __init__(self, action_space,percentage_alert=30,simu_step=0):
+        RecoPowerlineAgent.__init__(self, action_space)
+        self.percentage_alert = percentage_alert
+        self.simu_step=simu_step
+
+    def act(self, observation, reward, done=False):
+        action=super().act(observation, reward, done=False)
+        alertable_line_ids=observation.alertable_line_ids
+
+        #simu d'analyse de sécurité à chaque pas de temps sur les lignes attaquées
+        n_alertable_lines=len(alertable_line_ids)
+        nb_overloads=np.zeros(n_alertable_lines)
+        rho_max_N_1=np.zeros(n_alertable_lines)
+
+        # test which backend to know which method to call
+        N_1_actions=[self.action_space({"set_line_status": [(id_, -1)]}) for id_ in alertable_line_ids]
+        for i, action_to_simulate in enumerate(N_1_actions):
+
+            #check that line is not already disconnected
+            if (observation.line_status[alertable_line_ids[i]]):
+                (
+                    simul_obs,
+                    simul_reward,
+                    simul_has_error,
+                    simul_info,
+                ) = observation.simulate(action_to_simulate,time_step=self.simu_step)
```

Comment:
what do you do if your action is illegal ? It will then be equivalent to a game over ?
Beside, why not consider the action of the `controler` here in this simulate (so simulating `action_of_the_actor + powerline_disconnection` ? This would be more interesting I believe